components/VoteReasons.tsx

```diff
@@ -39,6 +40,29 @@ export function VoteReasons({ vote }: VoteReasonProps) {
     return proposals.some(p => p.id === vote.proposal.id);
   }, [proposals, vote.proposal.id]);
 
+  const publicClient = usePublicClient();
+  const [nnsName, setNnsName] = useState<string | null>(null);
+
+  useEffect(() => {
+    const getNnsName = async () => {
```

Comment:
seems like could factor this out to its own method/file. Maybe `lib/nns.ts`

Reply:
We could. Would you prefer that?